backends/ebpf/runtime/ebpf_runtime_test.c

```diff
@@ -0,0 +1,127 @@
+/*
+Copyright 2018 VMware, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include <unistd.h>     // getopt()
+#include <ctype.h>      // isprint()
+#include <string.h>     // memcpy()
+#include <stdlib.h>     // malloc()
+#include "ebpf_test.h"
+#include "ebpf_runtime_test.h"
+
+
+#define FILE_NAME_MAX 256
+#define MAX_10_UINT16 5;
+#define PCAPOUT "_out.pcap"
+
+/**
+ * @brief Feed a list packets into an eBPF program.
+ * @details This is a mock function emulating the behavior of a running
+ * eBPF program. It takes a list of input packets and iteratively parses them
+ * using the given imported ebpf_filter function. The output defines whether
+ * or not the packet is "dropped." If the packet is not dropped, its content is
+ * copied and appended to an output packet list.
+ *
+ * @param pkt_list A list of input packets running through the filter.
+ * @return The list of packets "surviving" the filter function
+ */
+pcap_list_t *feed_packets(packet_filter ebpf_filter, pcap_list_t *pkt_list, int debug) {
+    pcap_list_t *output_pkts = allocate_pkt_list();
+    uint32_t list_len = get_pkt_list_length(pkt_list);
+    for (uint32_t i = 0; i < list_len; i++) {
+        /* Parse each packet in the list and check the result */
+        struct sk_buff skb;
+        pcap_pkt *input_pkt = get_packet(pkt_list, i);
+        skb.data = (void *) input_pkt->data;
+        skb.len = input_pkt->pcap_hdr.len;
+        int result = ebpf_filter(&skb);
+        if (result != 0) {
+            /* We copy the entire content to emulate an outgoing packet */
+            pcap_pkt *out_pkt = copy_pkt(input_pkt);
+            output_pkts = append_packet(output_pkts, out_pkt);
+        }
+        if (debug)
+            printf("Result of the eBPF parsing is: %d\n", result);
+    }
+    return output_pkts;
+}
+
+/**
+ * @brief Create a pcap file name from a given base name, interface index,
+ * and suffix. Return value must be deallocated after usage.
+ * @param pcap_base The file base name.
+ * @param index The index of the file, represent an interface.
+ * @param suffix  Filename suffix (e.g., _in.pcap)
+ * @return An allocated string containing the file name.
+ */
+char *generate_pcap_name(const char *pcap_base, int index, const char *suffix) {
```

Comment:
why do you need two versions of this function?

Reply:
ebpf_runtime.c has a function that requires it, and so do ebpf_runtime_kernel.c/test.c. This is more a silly placement problem that came up once I moved a bunch of functions. Since code is duplicated in the kernel/test target anyway I just decided to duplicate it too, not super happy about it. I could put it in the pcap library. This solves the problem, but implies that we assume a specific file name pattern with our library.